src/VisualStudio/Core/Def/Implementation/Progression/GraphBuilder.cs

```diff
@@ -690,6 +691,98 @@ public GraphNode AddNodeForDocument(Document document)
             }
         }
 
+        public async Task<GraphNode> CreateNodeAsync(INavigateToSearchResult result, CancellationToken cancellationToken)
+        {
+            var document = result.NavigableItem.Document;
+            var project = document.Project;
+
+            // If it doesn't belong to a document or project we can navigate to, then ignore entirely.
+            if (document.FilePath == null || project.FilePath == null)
+                return null;
+
+            var category = result.Kind switch
+            {
+                NavigateToItemKind.Class => CodeNodeCategories.Class,
+                NavigateToItemKind.Delegate => CodeNodeCategories.Delegate,
+                NavigateToItemKind.Enum => CodeNodeCategories.Enum,
+                NavigateToItemKind.Interface => CodeNodeCategories.Interface,
+                NavigateToItemKind.Module => CodeNodeCategories.Module,
+                NavigateToItemKind.Structure => CodeNodeCategories.Struct,
+                NavigateToItemKind.Method => CodeNodeCategories.Method,
+                NavigateToItemKind.Property => CodeNodeCategories.Property,
+                NavigateToItemKind.Event => CodeNodeCategories.Event,
+                NavigateToItemKind.Constant or
+                NavigateToItemKind.EnumItem or
+                NavigateToItemKind.Field => CodeNodeCategories.Field,
```

Comment:
```suggestion
                NavigateToItemKind.Constant or
                    NavigateToItemKind.EnumItem or
                    NavigateToItemKind.Field => CodeNodeCategories.Field,
```

This is definitely a style preference, but I find it hard to know that items are linked to a single thing in switch expressions without the indentation.